design/Fancy calling conventions-design.md

```diff
@@ -0,0 +1,118 @@
+# CSE231 Final Project Design
+
+Members:
+
+- Lien-Bee Huang (lih004@ucsd.edu)
+- Xi-Zhen Liu (xil087@ucsd.edu)
+- Yang-Hang Wu (yaw038@ucsd.edu)
+
+## Interfaces
+
+### AST
+
+```typescript=
+
+export type Parameter<A> = { 
+    name: string, 
+    type: Type,
+    value: Expr<A>,
+}
+export type FunDef<A> = { 
+    a?: A, 
+    name: string, 
+    parameters: Array<Parameter<A>>, 
+    ret: Type, 
+    inits: Array<VarInit<A>>, 
+    body: Array<Stmt<A>>,
+    arbarg_idx: number,
+    kwarg_idx: number,
+}
+```
+
+To deal with parameters with default value, we would like to add an attribute call `value` in `Parameter<A>` to store the default value. It could be any expression at the parsing stage. If there's no specific default value, the attribute will be set to `null`. This can also help us distinguish `None` from no default value.
```

Comment:
Interesting! It could also be that it's a `?`-field, in which case checking if `value` is present tells us whether it's there.

Reply:
Helpful comment, we fixed our ast as your suggestion